tests/breed-hpprocurveN1178Test.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+require_once '../wwwroot/inc/deviceconfig.php';
+
+class hpprocurveN1178Test extends RTTestCase
+{
+
+	// hpprocurveN1178Read8021QConfig
+	/**
+	 * @dataProvider providerHP8021QConfigSamples
```

Comment:
The new method(s) should be annotated with `@group small`, otherwise the new tests are not run by `express.sh` during the automatic CI testing. You need to run specifically this test (or **all** tests, including heavy ones) to run this code:
```
$ phpunit --bootstrap bootstrap_v6v7.php breed-hpprocurveN1178Test.php
PHPUnit 6.5.5 by Sebastian Bergmann and contributors.

.
	OK! ----> HP 802.1Q Config Parser test case1 - parse interface brief.
	OK! ----> HP Show Interface Brief Parser test case1 .
	OK! ----> HP LLDP Parser test case1 - parse lldp data.
	OK! ----> HP Read MAC List Parser test case1 - all mac addresses
	OK! ----> HP Read MAC List Parser test case2 - mac addresses on port.                                                               5 / 5 (100%)
	OK! ----> HP 8021Q test case1 - access -> access: port 1 vlan 200 -> 201
	OK! ----> HP 8021Q test case2 - access -> trunk (+native): port 2 vlan 200 -> T200+201
	OK! ----> HP 8021Q test case3 - access -> trunk (-native): port 3 vlan 200 -> T+200,201
	OK! ----> HP 8021Q test case4 - trunk (+native) -> access: port 4 T200+201 -> access 200 (old native VID)
	OK! ----> HP 8021Q test case5 - trunk (+native) -> access: port 5 T200+201 -> access 201 (old tagged VID)
	OK! ----> HP 8021Q test case6 - trunk (+native) -> access: port 6 T200+201 -> access 202 (new VID)
	OK! ----> HP 8021Q test case7 - trunk (+native) -> trunk (+native): port 7 T200+201 -> T200+201-202 (add tagged vlan 202)
	OK! ----> HP 8021Q test case8 - trunk (+native) -> trunk (+native): port 7 T200+201-202 -> T200+201 (remove tagged vlan 202)
	OK! ----> HP 8021Q test case9 - trunk (+native) -> trunk (-native): port 8 T200+201 -> T+200-201 (keep old native 200 as tagged)
	OK! ----> HP 8021Q test case10 - trunk (+native) -> trunk (-native): port 9 T200+201 -> T+201 (remove old native 200 totally)
	OK! ----> HP 8021Q test case11 - trunk (+native) -> trunk (+native): port 10 T200+201 -> T200+202 (change tagged vlan to 202)
	OK! ----> HP 8021Q test case12 - trunk (-native) -> access: port 20 T+200-201 -> A202
	OK! ----> HP 8021Q test case13 - trunk (-native) -> trunk (+native): port 21 T+200-201 -> T200+201 (add native vlan 200)
	OK! ----> HP 8021Q test case14 - trunk (-native) -> trunk (-native): port 22 T+200-201 -> T+201 (remove tagged vlan 200)
	OK! ----> HP 8021Q test case15 - trunk (-native) -> trunk (-native): port 23 T+200-201 -> T+200-202 (add tagged vlan 202)
	OK! ----> HP 8021Q test case16 - trunk (-native) -> trunk (-native): port 24 T+200-201 -> T+202 (change all vlans to new tagged VID)
	OK! ----> HP 8021Q test case17 - get8021q
	OK! ----> HP 8021Q test case18 - getlldpstatus
	OK! ----> HP 8021Q test case19 - getportstatus
	OK! ----> HP 8021Q test case20 - getmaclist
	OK! ----> HP 8021Q test case21 - getportmaclist

Time: 164 ms, Memory: 10.00MB

OK (5 tests, 26 assertions)
```